src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp

```diff
@@ -498,6 +498,24 @@ bool crop_in_place_optimization::match(const program_node& node,
                 crop_params.input_offsets[0].spatial[1] != 0) {
                 return false;
             }
+            // Do not optimize out for possibly crop padding to select gemm_tiled_opt kernel
+            auto crop_layout = node.get_output_layout();
+            auto& crop_node = node.as<crop>();
+            std::pair<const program_node*, layout> user_info;
+            if (can_crop_be_optimized_simple_data_format(crop_layout, input_layout)) {
+                auto has_padding_expect_batch = [](const padding pad) {
+                    return std::any_of(std::next(pad._lower_size.begin()), pad._lower_size.end(), [](int32_t i){ return i > 0; }) ||
+                           std::any_of(std::next(pad._upper_size.begin()), pad._upper_size.end(), [](int32_t i){ return i > 0; });
+                };
+                update_in_place_crop_padding_simple_data_format(crop_layout,
+                                                                input_layout,
+                                                                user_info,
+                                                                crop_params.input_offsets[0],
+                                                                crop_node.get_primitive()->axis,
+                                                                is_runtime);
```

Comment:
* Did you check with onednn path? does onednn support padding in such case?
* I think this logic is weird because it is basically run-and-cancel. I think we can implement in a better way.

Reply:
PR updated.
Check with onednn path: OneDNN doesn't support padding. In last step of crop match func, it wouldn't optimize crop in case user's preferred impl type is onednn.